File: ui/src/lib/models/repo/Game.ts
export default interface Game {
    id: number;
    slug?: string;
    title: string;
    description: string;
    bannerImage: string;
    iconImage: string;
    gameCharacters: Array<GameCharacter>;
    releaseDate: Date;
    insertDate: Date;
}

export interface GameSeries {
    id: number;
    name: string;
    games: Array<Game>;
}

export interface GameCharacter {
    id: number;
    shortName: string;
    name: string;
    image: string;
    game: Game;
}

File: api/Data/Models/Game.cs
namespace TournamentWinner.Api.Models;

public class Game
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Slug { get; set; }
    public string Description { get; set; }
    public string BannerImage { get; set; }
    public string IconImage { get; set; }
    public ICollection<CharacterGame> CharacterGames { get; set; }
    public ICollection<CommunityGame> CommunityGames { get; set; }
    public DateTime ReleaseDate { get; set; }
    public DateTime InsertDate { get; set; }
}
